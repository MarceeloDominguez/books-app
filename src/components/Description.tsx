import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { COLOR } from "../constants/theme";

type Prop = {
  synopsis: string;
};

export default function Description({ synopsis }: Prop) {
  return (
    <View style={styles.content}>
      <Text style={styles.title}>Descripci√≥n</Text>
      <Text style={styles.synopsis}>{synopsis}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  content: {
    paddingVertical: 20,
  },
  title: {
    fontSize: 18,
    marginBottom: 10,
    letterSpacing: 0.3,
    color: COLOR.textPrimary,
    fontWeight: "bold",
  },
  synopsis: {
    fontSize: 15,
    letterSpacing: 0.3,
    color: COLOR.textPrimary,
    fontWeight: "600",
    lineHeight: 22,
  },
});
