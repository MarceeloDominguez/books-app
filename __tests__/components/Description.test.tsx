import renderer from "react-test-renderer";
import Description from "../../src/components/Description";
import { Text } from "react-native";

describe("Component Desciption", () => {
  it("verify that the component has rendered correctly", () => {
    const synopsis = "Esta es una sinopsis de prueba.";

    const component = renderer.create(<Description synopsis={synopsis} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should render synopsis correctly", () => {
    const synopsis = "Esta es una sinopsis de prueba.";

    const component = renderer.create(<Description synopsis={synopsis} />);

    const tree = component.root;

    //busco el elemento Text dentro del componente
    const textElements = tree.findAllByType(Text);

    //verifico que al menos haya un elemento text
    expect(textElements.length).toBeGreaterThan(0);

    //itero sobre los elementos Text y verifico si contiene la prop synopsis
    let found = false;
    textElements.forEach((textElement) => {
      if (textElement.props.children === synopsis) {
        found = true;
      }
    });

    expect(found).toBe(true);
  });
});
