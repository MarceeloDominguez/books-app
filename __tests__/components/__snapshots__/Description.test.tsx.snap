// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Desciption /> verify that the component has rendered correctly 1`] = `
<View
  style={
    {
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#191919",
        "fontSize": 18,
        "fontWeight": "bold",
        "letterSpacing": 0.3,
        "marginBottom": 10,
      }
    }
  >
    Descripci贸n
  </Text>
  <Text
    style={
      {
        "color": "#191919",
        "fontSize": 15,
        "fontWeight": "600",
        "letterSpacing": 0.3,
        "lineHeight": 22,
      }
    }
  >
    Esta es una sinopsis de prueba.
  </Text>
</View>
`;

exports[`Component Desciption verify that the component has rendered correctly 1`] = `
<View
  style={
    {
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#191919",
        "fontSize": 18,
        "fontWeight": "bold",
        "letterSpacing": 0.3,
        "marginBottom": 10,
      }
    }
  >
    Descripci贸n
  </Text>
  <Text
    style={
      {
        "color": "#191919",
        "fontSize": 15,
        "fontWeight": "600",
        "letterSpacing": 0.3,
        "lineHeight": 22,
      }
    }
  >
    Esta es una sinopsis de prueba.
  </Text>
</View>
`;

exports[`Desciption  1`] = `
<View
  style={
    {
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#191919",
        "fontSize": 18,
        "fontWeight": "bold",
        "letterSpacing": 0.3,
        "marginBottom": 10,
      }
    }
  >
    Descripci贸n
  </Text>
  <Text
    style={
      {
        "color": "#191919",
        "fontSize": 15,
        "fontWeight": "600",
        "letterSpacing": 0.3,
        "lineHeight": 22,
      }
    }
  >
    Esta es una sinopsis de prueba.
  </Text>
</View>
`;

exports[`Desciption verify that the component has rendered correctly 1`] = `
<View
  style={
    {
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#191919",
        "fontSize": 18,
        "fontWeight": "bold",
        "letterSpacing": 0.3,
        "marginBottom": 10,
      }
    }
  >
    Descripci贸n
  </Text>
  <Text
    style={
      {
        "color": "#191919",
        "fontSize": 15,
        "fontWeight": "600",
        "letterSpacing": 0.3,
        "lineHeight": 22,
      }
    }
  >
    Esta es una sinopsis de prueba.
  </Text>
</View>
`;
